content/app/models/file-path.js

```diff
@@ -0,0 +1,111 @@
+define("content/app/models/file-path", function(require, exports, module) {
+
+	/**
+	 * Simple class used for handling of file paths.
+	 */
+	var FilePath = Class({
+
+		$statics : {
+			REGEX_FOLDER_SEPARATOR : /\/|\)/g,
+			EXTENSION_SEPARATOR : '.',
+		},
+
+		/**
+		 * Class constructor
+		 * @param  	{String}	path 	Full file path.
+		 */
+		constructor : function(path) {
+			this._path = path;
+			this._name = null;
+			this._extension = null;
+			this._segments = null;
+			this._initialize();
+		},
+
+		/**
+		 * Name of the file without extension.
+		 * @type {String}
+		 */
+		name : {
+			get : function() {
+				return this._name;
+			}
+		},
+
+		/**
+		 * File extension (without leading '.');
+		 * @type {String}
+		 */
+		extension : {
+			get : function() {
+				return this._extension;
+			}
+		},
+
+		/**
+		 * File name with extension (example file.txt).
+		 * @type {String}
+		 */
+		nameWithExt : {
```

Comment:
This can be called filename, and it can be stored in _filename private variable instead of creating new string every time..